deploy/olm-catalog/smart-gateway-operator/manifests/smart-gateway-operator.clusterserviceversion.yaml

```diff
@@ -370,7 +370,7 @@ spec:
   - email: support@redhat.com
     name: Red Hat
   maturity: stable
-  minKubeVersion: 1.20.0
+  minKubeVersion: 1.23.0
```

Comment:
I asked wondered where this came from, and I suppose it's here: https://docs.openshift.com/container-platform/4.10/release_notes/ocp-4-10-release-notes.html#ocp-4-10-about-this-release
"This release uses [Kubernetes 1.23](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md)"

Reply:
If you also run `oc version` it'll tell you what the remote is running:

```
$  oc version
Client Version: 4.7.6
Server Version: 4.10.16
Kubernetes Version: v1.23.5+3afdacb
```

Reply:
actually I wonder if we need to change the minKubeVersion. Same about changing the 
LABEL com.redhat.openshift.versions="v4.10" with removing 4.7 and 4.8 from there. sgo *should* work with ocp 47 to 4.10, no?

Reply:
Technically not required, but I've been locking SGO and STO to specific supported releases linked to STO and don't plan to change that in this release. These will show up in a `stable-1.5` branch and another set of changes will happen where the base image changes beyond what we have for STF 1.4.

In theory we could make these changes in `stable-1.4` and then use that base branch for import to midstream and result in the artifacts used for STF 1.5. However just because of how everything is set up for our product pipelines, this really makes things much easier to manage and visualize in your head rather than having single branches that resolve into multiple pipelines. We used to do this with stf-common, and it ends up causing more releases than you'd otherwise have to do, along with potential to pull in new changes that aren't wanted as part of CVE builds etc.

So while this same code will work across multiple releases as you've stated, pipeline wise it's simpler in my experience to keep things locked to what STO will support. This release will be coming from a new `stable-1.5` channel in OLM as well, and eventually we'll stop doing releases for OCP 4.8 as they are going to be falling out of support relatively soon. OCP 4.7 is already EOL.

(This also keeps our test matrix smaller by not having to test against those older releases to verify that what we define as supported by keeping the minKubeVersion and olm.properties at a higher level than what the code could technically run on top of platform wise.)

Reply:
> sgo _should_ work with ocp 47 to 4.10, no?

I had this same thought. Leif's reply is insightful, but the a-ha moment for me yesterday was that (I think?) this change will only be on the master and stable-1.5 branches, and will not be backported to stable-1.4 branch (since it doesn't run on 4.10).

Reply:
Correct, this is only for OCP 4.10 and stable-1.5. There is no intent to backport any of the work slated for STF 1.5 to prior releases.

In this case, there is probably no _real_ reason to not scope this from 4.7 to 4.10, but I'm not aware of anyone using SGO independent of STF, and that would really be the only reason to have this bound differently from the release of STF it'll ship with. The related changes in STO will also only be available on v4.10 and later due to lack of availability of operators in prior releases to v4.10, so I'm not sure the larger test matrix is worth the effort.

One thing I was trying think of was whether there would be an advantage to having a larger support range for the purposes of upgrades. I don't see us having an administrator changing the OLM channel to stable-1.5 prior to upgrading their OCP 4.8 cluster to OCP 4.10 either.

I figure if we find that during the development of upgrade procedure from 4.8 to 4.10 (and subsequently STF 1.4 to 1.5) we find having things change to stable-1.5 on OCP 4.8 and then upgrading makes sense or provides a better UX we can always go back and change the range to something less restrictive. Primary motivation here is limiting our test matrix and that this code will be locked to a future `stable-1.5` branch and channel.

Reply:
I was not thinking of backporting. It was just: the limitation was unnecessary from the beginning.

I totally get it that it is not tested on later versions.